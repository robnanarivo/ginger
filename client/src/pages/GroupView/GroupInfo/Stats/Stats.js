import React from 'react';
import Stack from '@mui/material/Stack';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';

export default function TopicList(props) {
  const { groupInfo } = props;
  return (
    <Stack direction="row" spacing={6} sx={{ width: '100%' }}>
      <Box>
        <Typography sx={{ color: '#A9A9A9' }}> Created on</Typography>
        <Typography>
          {groupInfo.createOn}
        </Typography>
      </Box>
      <Box>
        <Typography sx={{ color: '#A9A9A9' }}> Members</Typography>
        <Typography>
          {groupInfo.member}
        </Typography>
      </Box>
    </Stack>
  );
}
