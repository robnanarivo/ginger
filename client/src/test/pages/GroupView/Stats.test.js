import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Stats from '../../../pages/GroupView/GroupInfo/Stats';
import 'jest-styled-components';

describe('Stats unit test', () => {
  it('Stats Snapshot test', () => {
    const tree = renderer
      .create(<Stats
        groupInfo={{ createOn: 'Feb 20 2008', member: '2.8k' }}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Stats contains correct element', () => {
    render(<Stats
      groupInfo={{ createOn: 'Feb 20 2008', member: '2.8k' }}
    />);
    // contains Created On heading
    expect(screen.getByText('Created on')).toBeInTheDocument();
    // contains Created On Content
    expect(screen.getByText('Feb 20 2008')).toBeInTheDocument();
    // contains Members heading
    expect(screen.getByText('Members')).toBeInTheDocument();
    // contains Members Content
    expect(screen.getByText('2.8k')).toBeInTheDocument();
  });
});
