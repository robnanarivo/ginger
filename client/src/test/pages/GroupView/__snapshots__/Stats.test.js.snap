// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats unit test Stats Snapshot test 1`] = `
<div
  className="css-12qb12d-MuiStack-root"
>
  <div
    className="MuiBox-root css-0"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ojz4j6-MuiTypography-root"
    >
       Created on
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Feb 20 2008
    </p>
  </div>
  <div
    className="MuiBox-root css-0"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ojz4j6-MuiTypography-root"
    >
       Members
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      2.8k
    </p>
  </div>
</div>
`;
